socorro/signature/rules.py

```diff
@@ -710,3 +710,43 @@ def action(self, raw_crash, processed_crash, notes):
             raw_crash['IPCMessageName']
         )
         return True
+
+
+class SignatureIPCMessageName(Rule):
+    """augments the signature if there is a IPC message name in the crash"""
+
+    def predicate(self, raw_crash, processed_crash):
+        return bool(raw_crash.get('IPCMessageName'))
+
+    def action(self, raw_crash, processed_crash, notes):
+        processed_crash['signature'] = '{} | IPC_Message_Name={}'.format(
+            processed_crash['signature'],
+            raw_crash['IPCMessageName']
+        )
+        return True
+
+
+class SignatureParentIDNotEqualsChildID(Rule):
+    """Stomp on the signature if MozCrashReason is parentBuildID != childBuildID
+
+    In the case where the assertion fails, then the parent buildid and the child buildid are
+    different. This causes a lot of strangeness particularly in symbolification, so the signatures
+    end up as junk. Instead, we want to bucket all these together so we replace the signature.
+
+    """
+
+    def predicate(self, raw_crash, processed_crash):
+        value = 'MOZ_RELEASE_ASSERT(parentBuildID == childBuildID)'
+        return bool(raw_crash.get('MozCrashReason', '') == value)
+
+    def action(self, raw_crash, processed_crash, notes):
+        notes.append(
+            'Signature replaced with MozCrashAssert, was: "%s"'
+            % processed_crash['signature']
+        )
+
+        # This is a failed assertion and the crash reason is the assertion that failed, so we put
+        # "!=" in the signature because that's what caused the assertion failure.
```

Comment:
I think it's clear what the line does without the comment. We can leave it in if you like anyway.

Reply:
I implemented this wrong the first time. I thought it looked sufficiently curious and wrong that it was prudent to add a verbose comment.